### ISSUE TITLE
IDEA: Awesome Miner integration

### ISSUE THREAD

**Issue by madmossy** [fireice-uk/xmr-stak#1404]
Integration with Awesome Miner would be great. API support within XMR-Stak so that it offers the same functionality as for example Claymore when used within the mining interface program Awesome Miner.

Allowing users complete control of XMR-Stak and the system (reboots, restarts, profit switching etc)

Currency its only possible to use XMR-Stak as an external miner with limit API access (can view hash rates and that's about all), this prevents using XMR-Stak for profit switching (switching between cryptonight pools/algorithms) and control of the miner itself, stop/start, restarting miner/system.

You would also need some way of passing configuration variables through the command line directly, rather than having to specify a config file for each coin or pool, gpu or mining rig you wanted to use.

Hope it would be possible to do this as I think XMR-Stak is by far the best miner currently for Cryptonight algorithms.

Cheers

**Comment by hbiyik** [comment]
This is related to awesome miner but, you can still add xmr stak as managed software, and then you can restart, stop, start xmrstak with awesome miner.  This is how i am doing it

https://www.reddit.com/r/MiningPoolHub/comments/7rq8g3/awesome_miner_xmrstak_electroneum_setup_figured/

**Comment by psychocrypt** [comment]
I know that this functions are maybe nice but on the other side those functionality also opens the doors for bot networks. What will definitive not come is "provit switching" because than you need a config per algorithm or coin. Bit this is currently very easy to handle by an extern tool which starts the miner with the correct config if it detects that an other coin is more profitable.

**Comment by bizlevel** [comment]
Security it is not problem, like on many nodes, use user and pass in config for Json Rest-API.
Also I use node on 127.0.0.1 with pass in config, and nginx + allow IP + iptables.
Triple security )

**Comment by Spudz76** [comment]
I agree on security, also forcing it to bind only to 127.0.0.1 is pretty safe too, let user redirect with proxy or iptables if they REALLY want it to be public accessible.

But it is not designed to "hot swap" even if sideloading new configs was enabled it would essentially exit and relaunch anyway / so just kill it and relaunch with external stuff?
